src/helpers/signingWallet.ts

```diff
@@ -104,3 +82,36 @@ export async function signWithSigningWallet(
   const signingWallet = new Wallet(decryptedPrivateKeyOfTheSigningWallet);
   return signingWallet.signMessage(messageToSign);
 }
+
+export async function createSignature(
+  address: EthereumAddress,
+  privateKey: string | null = null
+) {
+  logger.log('Creating a signature');
+  const publicKeyForTheSigningWallet = await getPublicKeyOfTheSigningWalletAndCreateWalletIfNeeded();
+
+  const mainWallet = privateKey
+    ? new Wallet(privateKey)
+    : await loadWallet(address, false);
+  if (mainWallet) {
+    const signatureForSigningWallet = await mainWallet.signMessage(
+      publicKeyForTheSigningWallet
+    );
+
+    const encryptor = new AesEncryptor();
+    const encryptedSignature = (await encryptor.encrypt(
+      RAINBOW_MASTER_KEY,
```

Comment:
I think it's outside of this PR, but I share a strong concern about any gain from that bi-directional encryption.